### PR TITLE
Apim 3111 adapt gio top bar

### PR DESCRIPTION
**Pull request by jourdiw**
**Issue**

https://github.com/gravitee-io/issues/issues/3111

**Description**

- Refactoring to centralize shared content / standardize variables
- Add oem customization to top bar 

Default: 

Screenshot 2023-11-02 at 16 55 26

Dark mode:

Screenshot 2023-11-02 at 16 56 02

Light mode:

Screenshot 2023-11-02 at 16 56 53

**Additional context**

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-policy-studio-angular
```
npm install @gravitee/ui-policy-studio-angular@7.42.0-apim-3111-adapt-gio-top-bar-4c8bd1b
```
```
yarn add @gravitee/ui-policy-studio-angular@7.42.0-apim-3111-adapt-gio-top-bar-4c8bd1b
```

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-particles-angular
```
npm install @gravitee/ui-particles-angular@7.42.0-apim-3111-adapt-gio-top-bar-4c8bd1b
```
```
yarn add @gravitee/ui-particles-angular@7.42.0-apim-3111-adapt-gio-top-bar-4c8bd1b
```

---

### CODE REVIEW

File: ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-submenu/gio-submenu.stories.ts [hunk 1]
```diff
@@ -13,19 +13,23 @@
...
-import { GioSubmenuGroupComponent } from './gio-submenu-group/gio-submenu-group.component';
 import { GioSubmenuModule } from './gio-submenu.module';
 
 export default {
   title: 'OEM Theme / Submenu',
   component: GioSubmenuGroupComponent,
   decorators: [
     moduleMetadata({
-      imports: [GioSubmenuModule],
+      imports: [GioSubmenuModule, NoopAnimationsModule, MatIconModule],
```
Comment (ThibaudAV) [inline ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-submenu/gio-submenu.stories.ts:32]: why MatIconModule is needed ?

File: ui-particles-angular/projects/ui-particles-angular/src/scss/index.scss [hunk 2]
```diff
@@ -19,6 +19,8 @@
 
 @forward './gio-oem-palette' show $mat-oem-palette, $mat-oem-palette-light;
 
+@forward './gio-oem-palette-variable';
```
[the remark this answers is not in the document]
Comment (jourdiw) [reply ui-particles-angular/projects/ui-particles-angular/src/scss/index.scss:22]: Nope we don't 🗑️

File: ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-top-bar/gio-top-bar-content/gio-top-bar-content.component.scss [hunk 3]
```diff
@@ -37,12 +38,21 @@ $typography: map.get(gio.$mat-theme, typography);
...
+    color: map.get(gio.$mat-oem-palette, background-contrast);
   }
 
   &.cockpit {
     background: rgb(135 111 236 / 40%);
     color: map.get(gio.$mat-accent-palette, default-contrast);
   }
 }
+
+::ng-deep {
```
Comment (ThibaudAV) [inline ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-top-bar/gio-top-bar-content/gio-top-bar-content.component.scss:51]: what about use :host-context(gio-top-bar-content) into gio-top-bar-menu ? 

in the world of angular components. it's up to the child to adapt to its environment, not the opposite :p

**Review by ThibaudAV — APPROVED** [review]

**Pull request merged**